### PR TITLE
[Interactive] some fixes in the prompt changer mechanism

### PR DESCRIPTION
**Pull request by ThomasFaivre**
Hi there!

I have encoutered some issues with the prompt changer mechanism:
 * doesn't work with IPython
 * requires a color theme change to effectively change the prompt

This PR fixes those issues and add 2 other related things:
 * make the theme module available in the startup script
 * add a prompt prefix configuration field (more info in the commitlog)

Sorry about trailing spaces in the first two patches, my editor fixes them automatically.

Thanks!

### CODE REVIEW

**Comment by guedou** [comment]
@p-l- it looks good to me.

File: scapy/config.py [hunk 1]
```diff
@@ -356,7 +388,8 @@ class Conf(ConfClass):
     checkIPinIP = True
     check_TCPerror_seqack = 0
     verb = 2
-    prompt = ">>> "
+    prompt = Interceptor("prompt", ">>> ", _prompt_changer)
```
Comment (gpotter2) [inline scapy/config.py]: Could you use `sys.ps1` ? And also replace it in `themes.py` ?

File: scapy/main.py [hunk 2]
```diff
@@ -373,31 +373,48 @@ def attr_matches(self, text):
...
             IPYTHON=False
-        
+
     if IPYTHON:
         banner = the_banner % (conf.version) + " using IPython %s" % IPython.__version__
 
+        from IPython.terminal.ipapp import load_default_config
+        from IPython.terminal import prompts
+
+        if conf.prompt == ">>> ":
```
Comment (gpotter2) [inline scapy/main.py]: Could you use `sys.ps1` ?

**Comment by ThomasFaivre** [comment]
@gpotter2 It looks like I can't use sys.ps1 because is not yet in interactive mode.
Therefore, there is no point in changing it in `main.py` and `config.py` but I could use `getattr` in `themes.py` as this might be called elsewhere. What do you think?

**Comment by gpotter2** [comment]
@ThomasFaivre What i wanted is to unify the pointers: it's not that good to have all those strings hardcoded.... Maybe every call could get the one in `config` ?

**Comment by gpotter2** [comment]
@ThomasFaivre Looks great :)

**Comment by gpotter2** [comment]
@ThomasFaivre Watch out: conflicts...

**Comment by gpotter2** [comment]
@guedou @ThomasFaivre Hi, all of this seems to have became very obsolete with the latest console changes :/ ThomasFaivre, you should give a look the new system (based mostly on IPython and without Pyreadline/Gnureadline)
Otherwise thi PR can be closed (at least for now)

**Comment by p-l-** [comment]
@ThomasFaivre closing this one based on @gpotter2 comment. Feel free to re-open or open another PR.

**Comment by ThomasFaivre** [comment]
Hi!

Some stuff are indeed obsolete but the issue still exists as I can't change the prompt:
```
>>> conf.prompt = 'toto'
>>> VERSION
'2.3.3.dev682'
```

I am looking into this!

**Comment by gpotter2** [comment]
@ThomasFaivre How about https://github.com/secdev/scapy/pull/824

**Pull request closed without merging**